src/magicbox/SuggestionsManager.ts

```diff
@@ -28,15 +35,46 @@ enum Direction {
   Right = 'Right'
 }
 
+export interface IPopulateSearchResultPreviewsEventArgs {
+  suggestionText: string;
+  previewsQuery: Promise<ISearchResultPreview[]>;
+}
+
+export enum SuggestionsManagerEvents {
+  PopulateSearchResultPreviews = 'populateSearchResultPreviews'
+}
+
 export class SuggestionsManager {
   public hasSuggestions: boolean;
   private pendingSuggestion: Promise<Suggestion[]>;
   private options: SuggestionsManagerOptions;
-  private keyboardFocusedSuggestion: HTMLElement;
+  private keyboardFocusedElement: HTMLElement;
   private suggestionsListbox: Dom;
   private suggestionsPreviewContainer: Dom;
   private lastSelectedSuggestion: HTMLElement;
   private root: HTMLElement;
+  private containsPreviews: boolean;
+  private previewContainer: Dom;
+  private lastPreviewsQuery: Promise<ISearchResultPreview[]>;
+
+  public get hasFocus() {
+    return $$(this.element).findClass(this.options.selectedClass).length > 0;
+  }
+
+  public get hasPreviews() {
+    return !!this.suggestionsPreviewContainer;
+  }
+
+  private get numberOfResultsPerRow() {
+    // To account for every CSS that may span previews over multiple rows, this solution was found: https://stackoverflow.com/a/49090306
```

Comment:
Is this comment still needed? I think the code on lines 74 and 75 is easy to follow.

Reply:
Good point, I'll remove this comment and I'll remove the TODOs.